src/exoticatechnologies/hullmods/exotics/HullmodExoticHandler.kt

```diff
@@ -0,0 +1,143 @@
+            )
+
+            // And obviously return true
+            true
+        }
+
+        AnonymousLogger.log("<-- shouldInstallHullmodExoticToVariant()\treturning ${retVal}", "HullmodExoticHandler")
+        return retVal
+    }
+
+
+    fun installHullmodExoticToVariant(hullmodExotic: HullmodExotic, parentFleetMember: FleetMemberAPI, variant: ShipVariantAPI): Boolean {
+        AnonymousLogger.log("--> installHullmodExoticToVariant()", "HullmodExoticHandler")
+        // Basically, this should be easy:
+        // 1. form up the key
+        // 2. grab the InstallData. Throw if there is none.
+        // 3. grab the already-installed list
+        // 4. throw if we're already in it
+        // 5. convert it to mutable list
+        // 6. add the variant to it
+        // 7. make a copy of the original lookup[key] value with the updated already-installed list
+        // 8. update the map with new installed data
+        val hullmodExoticKey = HullmodExoticKey(
+                hullmodExotic = hullmodExotic,
+                parentFleetMember = parentFleetMember
+        )
+
+        val currentInstallData = lookupMap[hullmodExoticKey]
+        if (currentInstallData == null) {
+            throw IllegalStateException("We should have had this key in the lookup map!!! lookupMap: ${lookupMap}")
+        }
+
+        val alreadyInstalledList = currentInstallData.listOfVariantsWeInstalledOn
+        // just check
+        val alreadyIn = alreadyInstalledList.contains(variant)
+        if (alreadyIn) {
+            // Actually, lets not throw just yet
+            // TODO change to throwing
+            AnonymousLogger.log("installHullmodExoticToVariant()\t\t !!!!! Trying to install on a variant that's already been installed on, returning false!!!!!", "HullmodExoticHandler")
+            return false
+        }
+        val mutableInstalledList = alreadyInstalledList.toMutableList()
+        mutableInstalledList.add(variant)
+        val newKeyValue = currentInstallData.copy(
+                parentFleetMemberAPI = currentInstallData.parentFleetMemberAPI,
+                listOfExpectedVariants = currentInstallData.listOfExpectedVariants,
+                listOfVariantsWeInstalledOn = mutableInstalledList.toList()
+        )
+        lookupMap[hullmodExoticKey] = newKeyValue
+
+        AnonymousLogger.log("<-- installHullmodExoticToVariant()\t\treturning true", "HullmodExoticHandler")
+        return true
+    }
+
+    fun grabAllKeysForParticularFleetMember(fleetMemberAPI: FleetMemberAPI): List<HullmodExoticKey> {
+        val retVal = mutableListOf<HullmodExoticKey>()
+        for(key in lookupMap.keys) {
+            if (key.parentFleetMember == fleetMemberAPI) retVal.add(key)
+        }
+
+        return retVal.toList()
+    }
+
+    fun getDataForKey(hullmodExoticKey: HullmodExoticKey): Optional<HullmodExoticInstallData> {
+        return Optional.ofNullable(lookupMap[hullmodExoticKey])
+    }
```

Comment:
these two should become private methods. and in the case for `getDataForKey` - it's gonna become highly unnecessary once `HullmodExoticHandler` starts doing **all** the work.

Reply:
they have been made private in #38 - resolving.